gamemode/modules/base/sh_util.lua

```diff
@@ -61,24 +61,54 @@ function DarkRP.findPlayer(info)
     if not info or info == "" then return nil end
     local pls = player.GetAll()
 
-    for k = 1, #pls do -- Proven to be faster than pairs loop.
-        local v = pls[k]
-        if tonumber(info) == v:UserID() then
-            return v
+    local count = #pls
+    local numberInfo = tonumber(info)
+    local lowerInfo = string.lower(tostring(info))
+
+    if numberInfo then -- since we'll be doing alot of scanning, need to make sure 
+        for k = 1, count do
+            local v = pls[k]
+
+            if numberInfo == v:UserID() then -- darkrp relies on this being first
+                return v
+            end
         end
 
-        if info == v:SteamID() then
-            return v
+        for k = 1, count do -- this loop could likely be combined with the above loop
```

Comment:
I don't think this loop _should_ be combined with the above loop, because then the order of players can influence the answer again